modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/src/Plugin/GCIdentityProvider/DaxkoSSO.php

```diff
@@ -151,6 +173,13 @@ public function submitConfigurationForm(array &$form, FormStateInterface $form_s
    */
   public function getLoginForm() {
 
+    if ($this->request->query->has('error')) {
+      return $this->formBuilder->getForm('Drupal\openy_gc_auth_daxko_sso\Form\TryAgainForm');
+    }
+    elseif ($this->configuration['login_mode'] === 'present_login_button') {
+      return $this->formBuilder->getForm('Drupal\openy_gc_auth_daxko_sso\Form\TryAgainForm');
```

Comment:
Should we redirect to a new login form in this case, but not to the Try again form?

Reply:
No. Try again form is the same for both cases. There is no need to copy this code.

Reply:
@anpolimus So the user sees the `Try again` button when comes to the page the first time. Is it ok? Should users see a login button instead?

Reply:
It's quite confusing to see `Try again` before you do something

Reply:
Every text on the button could be changed using form API.
We can store it at the config of the plugin method and give the ability for site.
Or we can change Try again text to: `Login with Daxko` - it could satisfy both cases.

Reply:
Or we can change text to `Login to Virtual Y` which is even more general and should work for both cases.

Reply:
ok, thx

Reply:
@dchubar-fj you were right.
It is better to do 2 different small forms for more flexible approach.
I'll do soon and ping you.

Reply:
It sounds like you're on the right track here. We discussed this in person a few days ago and I think it will be important to think about the flow for a non-active member here:

- Member gets to Daxko SSO login
- Member gets logged in to Daxko (if they don't they'll circle around inside Daxko's flow)
- Member returns to the VY site but is denied authorization.

At this point the first thing we should say to a member is "There's a problem with your account, AND here's how to get in touch to resolve it." The "Try again" option will not do anything until the member has contacted their branch, so we should ensure that the call to action is 1) contact us, 2) try again.